modules/phase_field/src/kernels/SoretDiffusion.C

```diff
@@ -14,6 +14,7 @@ InputParameters validParams<SoretDiffusion>()
   params.addRequiredCoupledVar("c", "Concentration");
   params.addRequiredParam<MaterialPropertyName>("diff_name", "The diffusivity used with the kernel");
   params.addParam<MaterialPropertyName>("Q_name", "Qheat", "The material name for the heat of transport");
+  params.addParam<FunctionName>("f_shape", "1.0", "The shape function found in denominator");
```

Comment:
We generally use MateriaProperties for such things (i.e. quantities that are _evaluated_ at quadrature points only).